Source/1.3/Territories/Territory.cs

```diff
@@ -65,69 +96,67 @@ public void UnclaimTile(int id)
             }
         }
 
+        /// <summary>
+        /// Claims all the tiles surrounding a <see cref="Settlement"/>. The radius is computed from <see cref="Faction"/>'s tech level.
+        /// This function does not check if the <see cref="RimWorld.Faction">settlement's owner</see> is the same as this
+        /// <see cref="RimWorld.Faction">territory's owner</see>.
+        /// </summary>
+        /// <param name="settlement">The settlement whose neighboring tiles should be claimed</param>
         public void SettlementClaimTiles(Settlement settlement)
         {
-            // This could cause a race condition where two Empires claim the same Tile
-            Task.Run(() => ClaimTiles(GetSurroundingTiles(settlement.Tile, (int)(faction.def.techLevel + 1))));
+            ClaimTiles(GetSurroundingTiles(settlement.Tile, (int)faction.def.techLevel + 1));
```

Comment:
Instead of removing the asynchronous call, maybe we should consider using a lock in the future.
Depends on how frequently settlements spawn, I suppose.

Reply:
We can, but from my benchmarking the function call takes <1ms (it prints 0ms) so I figured that since settlements aren't spawned that often, it shouldn't cause any lag. But yes, if we want to improve the tile claiming stuff (esp. figuring out race conditions and all) that might decrease performance.